builds/e2e/connectivity.yaml

```diff
@@ -177,14 +209,30 @@ jobs:
     variables:
       edgelet.artifact.name: 'iotedged-debian9-arm32v7'
     steps:
+      - task: Bash@3
+        name: Print_test_parameters
+        displayName: 'Print test parameters'
+        inputs:
+          targetType: inline
+          script: |
+            # print all matrix variables here
+            echo "testrun.config=$(testrun.config)"
+            echo "testrun.network.frequencies=$(testrun.network.frequencies)"
+            echo "testrun.network.runProfile=$(testrun.network.runProfile)"
+            echo "testrun.duration=$(testrun.duration)"
+            echo "deploymentFileName=$(deploymentFileName)"
+      - template: templates/set-run-flag.yaml
       - checkout: none
+        condition: and(succeeded(), eq(variables['run.steps'], 1))
       - task: AzureKeyVault@1
+        condition: and(succeeded(), eq(variables['run.steps'], 1))
         displayName: 'Azure Key Vault'
         inputs:
           azureSubscription: $(azure.subscription)
           KeyVaultName: $(azure.keyVault)
           SecretsFilter: 'edgebuilds-azurecr-io-username,edgebuilds-azurecr-io-pwd,EdgeConnectivityTestHubConnString,EdgeConnectivityEventHubConnString,kvLogAnalyticWorkspaceId,kvLogAnalyticSharedKey,EdgeConnectivityStorageAccountConnString'
       - task: DownloadBuildArtifacts@0
+        condition: and(succeeded(), eq(variables['run.steps'], 1))
```

Comment:
By default, `succeeded()` is a combination of `Succeeded` and `SucceededWithIssues` (continued w/ error)

Are you sure that is what you want there?

Reply:
this is exactly what we want as succeeded() will ensure all previous tasks succeeded.